recipe-client-addon/test/browser/browser_StudyStorage.js

```diff
@@ -0,0 +1,92 @@
+"use strict";
+
+Cu.import("resource://shield-recipe-client/lib/StudyStorage.jsm", this);
+
+// -------
+
+add_task(async function checkReset() {
+  ok(!StudyStorage.database, "StudyStorage starts without an IDB instance.");
+
+  await StudyStorage.getDatabase();
+  ok(StudyStorage.database, "StudyStorage creates an IDB instance.");
+
+  await StudyStorage.reset();
+  ok(!StudyStorage.database, "StudyStorage correctly removes its IDB instance.");
+});
+
+add_task(async function checkGetDatabase() {
+  await StudyStorage.reset(true);
```

Comment:
Nit: We can't really guarantee test execution order, so to avoid side-effects, we should clean up the study storage after the test is done rather than at the start of the test.

A `withStudyStorage` wrapper that automatically does this might be a good idea.